Add forgot password customizations for core profiler flow

ilyasfoo opened the pull request:
Related to #77829

## Proposed Changes

* Customize forgot password screens for WooCommerce core profiler flow
* Adds spinner during submitting email in forgot password
* Changes button copy from `Get Started` to `Log in` per design

## Testing Instructions

Adapted from https://github.com/Automattic/wp-calypso/pull/68558

In order to test, we'll need to set up a proxy using nginx the lost password API doesn't allow cross-domain POST.

**Setting up proxy**

1. Install nginx on your machine
2. Create an SSL cert and add it to your trusted store

```bash
# These commands are for OSX and Chrome.
# In a different system you may need to change the path for 
# /System/Library/OpenSSL/openssl.cnf and manually trust
# the generated certificate
 
# Generate the certificate
openssl req \
    -newkey rsa:2048 \
    -x509 \
    -nodes \
    -keyout ./config/server/key.pem \
    -new \
    -out ./config/server/certificate.pem \
    -subj /CN=wordpress.com \
    -reqexts SAN \
    -extensions SAN \
    -config <(cat /System/Library/OpenSSL/openssl.cnf \
     <(printf '[SAN]\nsubjectAltName=DNS:wpcalypso.wordpress.com')) \
    -sha256 \
    -days 365
 
# Add it to the trusted certificates
sudo security add-trusted-cert -d -k $(security list-keychains | grep System | cut -d '"' -f 2 ) ./config/server/certificate.pem
```

3. Add `127.0.0.1 wpcalypso.wordpress.com` to your `/etc/hosts` file
4. Use the following config for nginx (you can create new config in `/usr/local/etc/nginx/servers/default.ssl`) 
5. Start nginx

```nginx
server {
    listen 443 ssl;
    server_name wpcalypso.wordpress.com;

    ssl on;
    ssl_certificate <change_me>/wp-calypso/config/server/certificate.pem;
    ssl_certificate_key <change_me>/wp-calypso/config/server/key.pem;

    location / {
        proxy_pass https://wordpress.com;
        proxy_set_header Host $http_host;
    }

    location /calypso {
        proxy_pass http://127.0.0.1:3000;
    }

    location /cspreport {
       proxy_pass http://127.0.0.1:3000;
    }

    location /version {
        proxy_pass http://127.0.0.1:3000;
    }

    location /start {
        proxy_pass http://127.0.0.1:3000;
    }

    location /log-in {
        proxy_pass http://127.0.0.1:3000;
    }
}
```

6. Run `HOST=wpcalypso.wordpress.com yarn start`
7. Make sure you're logged out WordPress.com account.
8. Go to this [link](https://wpcalypso.wordpress.com/log-in/jetpack?redirect_to=http%3A%2F%2Fcalypso.localhost%3A3000%2Fjetpack%2Fconnect%2Fauthorize%3Fclient_id%3D190450138%26redirect_uri%3Dhttps%253A%252F%252Filyasfoowcpay.jurassic.tube%252Fwp-admin%252Fadmin.php%253Fhandler%253Djetpack-connection-webhooks%2526action%253Dauthorize%2526_wpnonce%253D3e526f1517%2526redirect%253Dhttps%25253A%25252F%25252Filyasfoowcpay.jurassic.tube%25252Fwp-admin%25252Fadmin.php%25253Fpage%25253Dwc-admin%26state%3D1%26scope%3Dadministrator%253A03762e316bb79e73117c81a2670dabfd%26user_email%3Dwordpress%2540example.com%26user_login%3Dnrzo%26jp_version%3D12.3-a.6%26secret%3DK8HGrfxbBt6yMpfnx0h8RO6bdKu3pU3a%26blogname%3DHelloWord%2526%2523039%253Bs%26site_url%3Dhttps%253A%252F%252Filyasfoowcpay.jurassic.tube%26home_url%3Dhttps%253A%252F%252Filyasfoowcpay.jurassic.tube%26site_icon%3D%26site_lang%3Den_US%26site_created%3D2023-06-12%252005%253A44%253A13%26allow_site_connection%3D1%26_as%3Dunknown%26from%3Dwoocommerce-core-profiler%26calypso_env%3Dproduction%26_ui%3D189375772%26_ut%3Dwpcom%253Auser_id%26purchase_nonce%3DZyjFK3lz%26_wp_nonce%3D9fcf83c152%26redirect_after_auth%3Dhttps%253A%252F%252Filyasfoowcpay.jurassic.tube%252Fwp-admin%252Fadmin.php%253Fpage%253Dwc-admin%26site%3Dhttps%253A%252F%252Filyasfoowcpay.jurassic.tube&email_address=aegisrunestone3%40gmail.com&from=woocommerce-core-profiler)
9. Click on `Forgot password`
10. Observe the screen matches the first screenshot below
11. Enter your email address and click on `Reset my password`
12. Observe spinner loading
13. Observe you're redirect to a screen that matches the second screenshot below
14. You should receive an email to reset your password

<img width="1224" alt="image" src="https://github.com/Automattic/wp-calypso/assets/3747241/7751a345-c3d0-4aa9-8bc2-c5c4a0e821d1">

<img width="1224" alt="image" src="https://github.com/Automattic/wp-calypso/assets/3747241/7081ba14-635f-4448-a6ea-82c08f5673e0">

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

ilyasfoo committed:
Add forgot password customizations for core profiler flow, added loading spinner for lost password button

chihsuan commented on client/blocks/login/lost-password-form.jsx:
```diff
@@ -1,4 +1,4 @@
-import { FormInputValidation } from '@automattic/components';
+import { FormInputValidation, Spinner } from '@automattic/components';
```
Hey @ilyasfoo I think we should use `Spinner` from `@wordpress/components` (Y5pUYSJPsGEud1vknUZhi8-fi-1749_35798). 🙂

ilyasfoo replied:
Good catch! I've replaced it in 7352c79420

chihsuan commented on client/layout/masterbar/woo.scss:
```diff
@@ -873,5 +873,39 @@
+		.login__lostpassword-form {
+			display: flex;
+			flex-direction: column;
+			padding: 54px 24px 32px;
```
It seems that the padding-top should be set to `48px`?

![Screenshot 2023-06-20 at 11 45 44](https://github.com/Automattic/wp-calypso/assets/4344253/5d017d7b-f59b-4678-b00b-c780320f9fdd)

ilyasfoo replied:
Good catch! Fixed in 7352c79420

chihsuan reviewed (commented):
Thanks for working on this! @ilyasfoo Looks great. 👍 

Some minor UI feedback:

1. We should set `Signup link` color to the theme color. It's Woo purple now.
2. Heading title line height should be `32px` 
3. Form button margin-top should be `24px` ([Screenshot](https://github.com/Automattic/wp-calypso/assets/4344253/617afea5-9a9b-4942-8d94-459b28412c7e))
4. Label margin-bottom should be `16px` ([Screenshot](https://github.com/Automattic/wp-calypso/assets/4344253/9e55e8a1-0dd4-4107-9244-78c9ec523b0c)). It's `20px` now. Also, should we change the label name on the rest password screen to `your email address`, instead of `YOUR EMAIL ADDRESS OR USERNAME`

chihsuan commented:
> In order to test, we'll need to set up a proxy using nginx the lost password API doesn't allow cross-domain POST.

Thanks, we can also use our [start dev env](https://github.com/woocommerce/woocommerce-start-dev-env) as well. 🙂  cc @rjchow

rjchow commented:
> > In order to test, we'll need to set up a proxy using nginx the lost password API doesn't allow cross-domain POST.
> 
> Thanks, we can also use our [start dev env](https://github.com/woocommerce/woocommerce-start-dev-env) as well. 🙂 cc @rjchow

Confirm that reset password works correctly with start-dev-env

rjchow reviewed (approved):
Pre-approving, conditional on addressing Chi's UI feedback
It worries me a bit that the code for our flow is so tightly intertwined with the other flows, but I'm not sure much can be done about that. Mostly concerned about a repeat of the incident we had recently where a change by another team broke the new user sign up

ilyasfoo committed:
Replaced spinner component and adjust forgot password form margin

ilyasfoo committed:
Address styling fixes

ilyasfoo committed:
Change email address label copy

ilyasfoo commented:
Thanks for the review, @chihsuan @rjchow!

> 1. We should set `Signup link` color to the theme color. It's Woo purple now.
> 2. Heading title line height should be `32px` 
> 4. Label margin-bottom should be `16px` ([Screenshot](https://github.com/Automattic/wp-calypso/assets/4344253/9e55e8a1-0dd4-4107-9244-78c9ec523b0c)). It's `20px` now. Also, should we change the label name on the rest password screen to `your email address`, instead of `YOUR EMAIL ADDRESS OR USERNAME`

Addressed these in 8174de3 and 6d54b18

> 3. Form button margin-top should be 24px ([Screenshot](https://github.com/Automattic/wp-calypso/assets/4344253/617afea5-9a9b-4942-8d94-459b28412c7e))

It's already set to 24px, let me know if you're finding it to be different:

<img width="671" alt="image" src="https://github.com/Automattic/wp-calypso/assets/3747241/59af6a9a-3a3e-4f68-b267-3a617b64b6ef">

ilyasfoo commented:
> Thanks, we can also use our [start dev env](https://github.com/woocommerce/woocommerce-start-dev-env) as well. 🙂 cc @rjchow

Ah, I totally forgot about using start dev env! Thanks for reminding me, I just used it for addressing the feedback.

ilyasfoo committed:
Fix link line height

chihsuan reviewed (approved):
Thanks @ilyasfoo Looks great! 🚀 

> It's already set to 24px, let me know if you're finding it to be different:

Ah, yes, it's 24px. 🙌

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8061445

Thank you @ilyasfoo for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8061445).
